snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedRequests.java

```diff
@@ -386,5 +397,97 @@ public static SnomedRepositoryCommitRequestBuilder prepareCommit() {
 	public static SnomedConceptSearchRequestBuilder prepareGetSynonyms() {
 		return prepareSearchConcept().all().filterByActive(true).filterByEcl("<<"+Concepts.SYNONYM);
 	}
-
+	
+	public static Promise<Collection<String>> getApplicableTypes(final IEventBus bus, final String resourcePath, 
+			final Set<String> ruleParentIds, 
+			final Set<String> refSetIds,
+			final List<String> moduleIds,
+			boolean needsDataAttributes,
+			boolean needsObjectAttributes) {
+		return SnomedRequests.prepareSearchMember()
+			.all()
+			.filterByActive(true)
+			.filterByRefSetType(SnomedRefSetType.MRCM_MODULE_SCOPE)
+			.filterByReferencedComponent(moduleIds)
+			.build(resourcePath)
+			.execute(bus)
+			.then(members -> members.stream()
+					.map(m -> (String) m.getProperties().get(SnomedRf2Headers.FIELD_MRCM_RULE_REFSET_ID))
+					.collect(Collectors.toSet()))
+			.thenWith(inScopeRefSetIds -> {
+				Set<String> domainIds = new HashSet<>();
+				
+				if (ruleParentIds != null) {
+					domainIds.addAll(ruleParentIds);
+				}
+				
+				if (refSetIds != null) {
+					domainIds.addAll(refSetIds);
+				}
+				
+				return SnomedRequests.prepareSearchMember()
+					.all()
+					.filterByActive(true)
+					.filterByRefSetType(SnomedRefSetType.MRCM_ATTRIBUTE_DOMAIN)
+					.filterByRefSet(inScopeRefSetIds)
+					.filterByProps(Options.from(Map.of(SnomedRf2Headers.FIELD_MRCM_DOMAIN_ID, domainIds)))
+					.build(resourcePath)
+					.execute(bus);
+			}).thenWith(members -> {
+				Set<String> typeIds = members.stream().map(m -> m.getReferencedComponentId()).collect(Collectors.toSet());
+				if ((needsDataAttributes && needsObjectAttributes) || typeIds.isEmpty()) {
```

Comment:
What happens if I specify both boolean parameters with the value `false`?

Reply:
Changed this to use an enum instead, so this wouldn't happen.